.github/workflows/libcxx-build-and-test.yaml

```diff
@@ -0,0 +1,137 @@
+name: Build and Test libc++
+
+on: pull_request
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+permissions:
+  checks: write
+
+env:
+  CMAKE: "/opt/bin/cmake"
+
+# Comment
+jobs:
+  stage1:
+    runs-on: libcxx-runners-16
+    continue-on-error: false
+    strategy:
+      fail-fast: true
+      matrix:
+        config: [ 'generic-cxx26', 'generic-cxx03', 'generic-modules' ]
+        cc: [ 'clang-18' ]
+        cxx: [ 'clang++-18' ]
+        include:
+          - config: 'generic-gcc'
+            cc: 'gcc-13'
+            cxx: 'g++-13'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 5
```

Comment:
Is there a reason the fetch depth here (and below) is 5? As far as I understand, building the code should just need the top commit (although fetching 5 vs 1 is a negligible time impact in clone time).

Reply:
This can go. I was attempting to do a thing with git diff, but that thing is long gone.